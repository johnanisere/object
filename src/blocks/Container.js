import styled from "styled-components";

const Container = styled.section`
  display: flex;
  padding: 20px 10px;
  @media (max-width: 768px) {
    flex-direction: column;
  }
`;
export default Container;
