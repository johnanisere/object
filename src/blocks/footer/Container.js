import styled from "styled-components";

const Container = styled.div`
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  text-align: ${({ align }) => align || "left"};
  flex: ${({ flex }) => (flex ? flex : 1)};
`;

export default Container;
