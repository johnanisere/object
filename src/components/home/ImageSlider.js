import React from "react"
import PropTypes from "prop-types"
import Image from "../../blocks/Image"
import Container from "../../blocks/FlexibleContainer";


const App = ({ data, onSwitchImage, active }) => {
    return (
        <Container background={true} slider={true}>
            <button>{`<`}</button>
            <div>
                {
                    data.map(({ displayImage }, key) => <Image onClick={() => onSwitchImage(key)} active={key !== active} key={key}><img alt="me" src={displayImage} /></Image>)
                }
            </div>
            <button>></button>
        </Container>
    )
}

App.propTypes = {
    data: PropTypes.array.isRequired,
    active: PropTypes.number.isRequired,
    onSwitchImage: PropTypes.func.isRequired
}
export default App