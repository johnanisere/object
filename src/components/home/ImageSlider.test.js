
import React from "react";
import { shallow } from "enzyme";

import ImageSlider from "./ImageSlider"
import Container from "../../blocks/FlexibleContainer";


const props = {
    data: [],
    active: 1,
    onSwitchImage: () => { }
}

it("ImageSlider component exists", () => {
    const wrapper = shallow(<ImageSlider {...props} />);
    expect(wrapper.exists()).toBe(true);
});

it("ImageSlider component renders", () => {
    const wrapper = shallow(<ImageSlider {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it(`ImageSlider should contain "Container",  "ImageSlider", `, () => {
    const wrapper = shallow(<ImageSlider {...props} />);

    expect(wrapper.find('button')).toHaveLength(2);
    expect(wrapper.find('div')).toHaveLength(1);
    expect(wrapper.find(Container)).toHaveLength(1);
});