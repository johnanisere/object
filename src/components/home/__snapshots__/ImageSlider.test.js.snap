// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider component renders 1`] = `ShallowWrapper {}`;
